File: Api/Model/Advisor/BuyRequest.cs
using Api.Model.Account;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Api.Model.Advisor
{
    public class BuyRequest
    {
        public int PortfolioId { get; set; }
        public int Days { get; set; }
        public string Address { get; set; }
        public SetGoalRequest Goal { get; set; }
    }
}

File: Web/src/app/model/advisor/buyRequest.ts
import { Goal } from "../account/goal";

export class BuyRequest {
    portfolioId: number;
    days : number;
    address : string;
    invested : number;
    goal : Goal;

    constructor(portfolioId: number, days: number, address: string, invested : number, goal?: Goal){
      this.portfolioId = portfolioId;
      this.days = days;
      this.address = address;
      this.invested = invested;
      this.goal = goal;
    }
  }
  